ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -156,157 +246,92 @@ public CompletableFuture<NamedBlobRecord> get(String accountName, String contain
+          try {
+            run_put_v2(record, accountId, containerId, connection);
+          } catch (Exception e) {
+            logger.error("NAMED PUT: Data Error: old table succeed while new table failed: ", e);
+            metricsRecoder.namedDataErrorPutCount.inc();
           }
-          return new PutResult(record);
+          metricsRecoder.namedBlobPutTimeInMs.update(System.currentTimeMillis() - startTime);
+          return putResult;
         }, null);
   }
 
   @Override
   public CompletableFuture<DeleteResult> delete(String accountName, String containerName, String blobName) {
     return executeTransactionAsync(accountName, containerName, false, (accountId, containerId, connection) -> {
-      String blobId;
-      Timestamp currentTime;
-      boolean alreadyDeleted;
-      try (PreparedStatement statement = connection.prepareStatement(SELECT_FOR_SOFT_DELETE_QUERY)) {
-        statement.setInt(1, accountId);
-        statement.setInt(2, containerId);
-        statement.setString(3, blobName);
-        try (ResultSet resultSet = statement.executeQuery()) {
-          if (!resultSet.next()) {
-            throw buildException("DELETE: Blob not found", RestServiceErrorCode.NotFound, accountName, containerName,
-                blobName);
-          }
-          blobId = Base64.encodeBase64URLSafeString(resultSet.getBytes(1));
-          Timestamp deletionTime = resultSet.getTimestamp(2);
-          currentTime = resultSet.getTimestamp(3);
-          alreadyDeleted = (deletionTime != null && currentTime.after(deletionTime));
+      long startTime = System.currentTimeMillis();
+      DeleteResult deleteResult, deleteResultV2;
+      deleteResult = run_delete(accountName, containerName, blobName, accountId, containerId, connection);
+      try {
+        deleteResultV2 = run_delete_v2(accountName, containerName, blobName, accountId, containerId, connection);
+        if (!deleteResult.equals(deleteResultV2)) {
+          logger.warn("NAMED DELETE: Data Inconsistence: old record='{}', new record='{}'", deleteResult, deleteResultV2);
+          metricsRecoder.namedDataInconsistentDeleteCount.inc();
         }
-      }
-      // only need to issue an update statement if the row was not already marked as deleted.
-      if (!alreadyDeleted) {
-        try (PreparedStatement statement = connection.prepareStatement(SOFT_DELETE_QUERY)) {
-          // use the current time
-          statement.setTimestamp(1, currentTime);
-          statement.setInt(2, accountId);
-          statement.setInt(3, containerId);
-          statement.setString(4, blobName);
-          statement.executeUpdate();
+      } catch (Exception e) {
+        if (e instanceof RestServiceException &&
+            ((RestServiceException) e).getErrorCode() == RestServiceErrorCode.NotFound) {
+          logger.warn("NAMED DELETE: Data Not Found: old table succeed while new table got NotFound error: ", e);
```

Comment:
let's not add a warn log message here, since just like get, this might happen a lot before backfill.

Reply:
Done in latest commit.